File: api/src/Confluence.Domain/Entities/TradeReview.cs
using Confluence.Domain.Enums;

namespace Confluence.Domain.Entities;

public class TradeReview
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid TradeId { get; set; }
    
    // Overall review verdict: "good", "fair", "poor"
    public string Verdict { get; set; } = string.Empty;
    // 0.0 – 1.0 execution quality score
    public decimal ExecutionScore { get; set; }
    
    // AI-generated structured review
    public bool PlanAdherence { get; set; }
    public string PlanAdherenceExplanation { get; set; } = string.Empty;
    
    public bool SlTpRational { get; set; }
    public string SlTpAnalysis { get; set; } = string.Empty;
    
    public string TimingVerdict { get; set; } = string.Empty;
    public string TimingExplanation { get; set; } = string.Empty;
    
    public string ImprovementAdvice { get; set; } = string.Empty;
    
    // Full AI output for reference/debugging
    public string FullReviewJson { get; set; } = string.Empty;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    // Navigation
    public Trade? Trade { get; set; }
}

File: ai-service/app/schemas/trade_review.py
from pydantic import BaseModel, Field

class TradeReviewRequest(BaseModel):
    trade_id: str = Field(..., description="Trade UUID")
    symbol: str = Field(..., description="Trading pair, e.g. 'BTC/USDT'")
    direction: str = Field(..., description="'Long' or 'Short'")
    entry_price: float
    exit_price: float
    stop_loss: float | None
    take_profit: float | None
    leverage: float
    entry_at: str
    exit_at: str | None
    pnl_quote: float | None
    pnl_percentage: float | None
    tags: str | None
    notes: str | None
    analysis_result_json: str | None = Field(None, description="Original AI analysis at trade entry time")

class TradeReviewResponse(BaseModel):
    verdict: str = Field(..., description="'good', 'fair', or 'poor'")
    execution_score: float = Field(..., ge=0.0, le=1.0, description="0.0 to 1.0 execution quality score")
    plan_adherence: bool = Field(..., description="Did the trade follow the original AI plan?")
    plan_adherence_explanation: str = Field(..., description="Explanation of plan adherence")
    sl_tp_rational: bool = Field(..., description="Were the SL and TP levels technically rational?")
    sl_tp_analysis: str = Field(..., description="Analysis of the SL and TP levels")
    timing_verdict: str = Field(..., description="'early_exit', 'late_exit', or 'optimal'")
    timing_explanation: str = Field(..., description="Explanation of the exit timing")
    improvement_advice: str = Field(..., description="Advice for future trades with similar setups")
